src/ArchiveManagement/NexusMods.FileExtractor/Extractors/SevenZipExtractor.cs

```diff
@@ -221,9 +228,24 @@ private static string GetExtractorExecutable(IFileSystem fileSystem)
     /// See https://github.com/Nexus-Mods/NexusMods.App/issues/1306 for details.
     /// </remarks>
     private const bool UseSystemExtractor =
-#if USE_SYSTEM_EXTRACTOR
+#if NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR || USE_SYSTEM_EXTRACTOR
```

Comment:
There should only be one constant. Either `NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR` or `USE_SYSTEM_EXTRACTOR`, not both.

Reply:
Speaking as a downstream packager, I'm fine with these kinda changes being done in a breaking way so long as the docs get updated and the change is mentioned in release notes.

Packagers shouldn't be surprised when experimental software has breaking changes :grinning:

Reply:
There should've only been one compile constant in the beginning, which is `NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR`. I still don't understand what caused #1836.

Reply:
> There should've only been one compile constant in the beginning, which is `NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR`.

Ah yes, that's what I was using. I got confused as the diff made it look like `NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR` was a new constant replacing `USE_SYSTEM_EXTRACTOR`. Should've double checked before commenting!

I don't fully understand the solution, but this seems relevant https://github.com/Nexus-Mods/NexusMods.App/pull/1919#issuecomment-2303284678:
> So we can't do an alias by appending `DefineConstants` in the `.csproj` or `.targets` file. Instead I alias by checking both define names

Reply:
@erri120 themselves originally added both USE_SYSTEM_EXTRACTOR and NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR.

Due to how they were used in the `.targets` file, I assumed it was meant to be an alias.
I kept this alias in the PR; should I not have done that?

AFAIK downstreams only use `NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR` and not `USE_SYSTEM_EXTRACTOR`, so the latter is safe to remove.

Reply:
> Setting `DefineConstants` via CLI acts as an 'absolute override', i.e. any changes you make to them, in `.csproj`, `.targets` or otherwise get discarded.

Possibly out of scope for this PR, but is that a possible justification for moving to the pattern described in https://github.com/NixOS/nixpkgs/pull/331150#discussion_r1727635149?

I.e:
```xml
  <PropertyGroup Condition="$(UseSystemExtractor) == 'true'">
    <DefineConstants>$(DefineConstants);NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR</DefineConstants>
  </PropertyGroup>
```

Paired with `dotnet build -p:UseSystemExtractor=true`.

Reply:
Personally I would prefer it that way, after (re)discovering more of the nature in how `DefineConstants` interacts with the rest of the build system.